packages/core-data/src/resolvers.js

```diff
@@ -261,37 +261,41 @@ export const getEntityRecords =
 				} );
 			}
 
-			dispatch.receiveEntityRecords(
-				kind,
-				name,
-				records,
-				query,
-				false,
-				undefined,
-				meta
-			);
+			registry.batch( () => {
+				dispatch.receiveEntityRecords(
+					kind,
+					name,
+					records,
+					query,
+					false,
+					undefined,
+					meta
+				);
 
-			// When requesting all fields, the list of results can be used to
-			// resolve the `getEntityRecord` selector in addition to `getEntityRecords`.
-			// See https://github.com/WordPress/gutenberg/pull/26575
-			if ( ! query?._fields && ! query.context ) {
-				const key = entityConfig.key || DEFAULT_ENTITY_KEY;
-				const resolutionsArgs = records
-					.filter( ( record ) => record[ key ] )
-					.map( ( record ) => [ kind, name, record[ key ] ] );
+				// When requesting all fields, the list of results can be used to
+				// resolve the `getEntityRecord` selector in addition to `getEntityRecords`.
+				// See https://github.com/WordPress/gutenberg/pull/26575
+				if ( ! query?._fields && ! query.context ) {
+					const key = entityConfig.key || DEFAULT_ENTITY_KEY;
+					const resolutionsArgs = records
+						.filter( ( record ) => record[ key ] )
+						.map( ( record ) => [ kind, name, record[ key ] ] );
+
+					dispatch( {
+						type: 'START_RESOLUTIONS',
+						selectorName: 'getEntityRecord',
+						args: resolutionsArgs,
+					} );
+					dispatch( {
+						type: 'FINISH_RESOLUTIONS',
+						selectorName: 'getEntityRecord',
+						args: resolutionsArgs,
+					} );
+				}
 
-				dispatch( {
-					type: 'START_RESOLUTIONS',
-					selectorName: 'getEntityRecord',
-					args: resolutionsArgs,
-				} );
-				dispatch( {
-					type: 'FINISH_RESOLUTIONS',
-					selectorName: 'getEntityRecord',
-					args: resolutionsArgs,
-				} );
-			}
-		} finally {
+				dispatch.__unstableReleaseStoreLock( lock );
+			} );
+		} catch ( e ) {
```

Comment:
I think we should switch this back to `try...finally`. The catching error here means it won't be passed to the resolution state (`getResolutionError`) and might affect the error state for `useSuspenseSelect`. (#37261).

But that means we won't be able to batch `dispatch.__unstableReleaseStoreLock`.

https://github.com/WordPress/gutenberg/blob/0d1943cf84e60ad4646333ed68eca7cc7b879009/packages/data/src/redux-store/index.js#L533-L535